compiler/rustc_mir_dataflow/src/impls/liveness.rs

```diff
@@ -137,23 +138,26 @@ impl DefUse {
         match context {
             PlaceContext::NonUse(_) => None,
 
-            PlaceContext::MutatingUse(MutatingUseContext::Store | MutatingUseContext::Deinit) => {
-                Some(DefUse::Def)
-            }
+            PlaceContext::MutatingUse(
+                MutatingUseContext::Store
+                | MutatingUseContext::Deinit
+                | MutatingUseContext::AsmOutput,
+            ) => Some(DefUse::Def),
 
             // Setting the discriminant is not a use because it does no reading, but it is also not
             // a def because it does not overwrite the whole place
-            PlaceContext::MutatingUse(MutatingUseContext::SetDiscriminant) => None,
+            PlaceContext::MutatingUse(MutatingUseContext::SetDiscriminant) => {
+                Some(DefUse::MaybeDef)
+            }
 
             // `MutatingUseContext::Call` and `MutatingUseContext::Yield` indicate that this is the
             // destination place for a `Call` return or `Yield` resume respectively. Since this is
             // only a `Def` when the function returns successfully, we handle this case separately
-            // in `call_return_effect` above.
-            PlaceContext::MutatingUse(
-                MutatingUseContext::Call
-                | MutatingUseContext::AsmOutput
```

Comment:
Why `AsmOutput` change? The assembly output case seems analogous to a call and a yield, with a local being defined only after returning from the assembly block.

Reply:
Oh, for some reason I was under the impression that `TerminatorKind::InlineAsm` had no cleanup path... not sure what got into me there. Will fix